### PR TITLE
Introduce llama3-java module

### PR DESCRIPTION
**Pull request by geoand**
- Resolves: #997

### CODE REVIEW

File: model-providers/llama3-java/pom.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        <artifactId>quarkus-langchain4j-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+    <artifactId>quarkus-langchain4j-llama3-java-parent</artifactId>
+    <name>Quarkus LangChain4j - Llama 3 - Java - Parent</name>
+    <packaging>pom</packaging>
+
+    <properties>
+        <maven.compiler.release>21</maven.compiler.release>
```
Comment (edeandrea) [inline model-providers/llama3-java/pom.xml:15]: Really a Java 21 baseline?

Comment (geoand) [reply model-providers/llama3-java/pom.xml:15]: What's the alternative?

Comment (edeandrea) [reply model-providers/llama3-java/pom.xml:15]: 17?

Comment (geoand) [reply model-providers/llama3-java/pom.xml:15]: Vector support on 17? Don't think so...

Comment (edeandrea) [reply model-providers/llama3-java/pom.xml:15]: Ahhhhh....You are right!

I was only looking at this from a JVM baseline - where Quarkus core baselines on 17. But you are right.

**Comment by geoand** [comment]
This still needs at least the following:

 * Download models at build time
 * Allow copying the models into the fast-jar like we did for Jlama
 * Fix the build to ensure that the preview features used does not prevent the extension from being used in JDKs other than the target version


Ideally we should also have some docs about how to use this and Jlama. @mariofusco IIRC you wanted to write something about that?

**Comment by geoand** [comment]
Awesome, that would go a long way to helping people experiment with these two great technologies!

**Comment by geoand** [comment]
This should now be good enough for basic usage. 

In JVM mode, one would build the application with:

```
mvn package
```

and run it like so:


```
java --enable-preview -jar integration-tests/llama3-java/target/quarkus-app/quarkus-run.jar
``` 

For native mode, best use the latest GraalVM from sdkman (I used `sdk install java 24.ea.18-graal`)

and build using:

```
mvn package -Dnative
```

but keeping it mind that you probably want to have:

```properties
quarkus.native.march=native
quarkus.native.additional-build-args=-O3
```

in `application.properties` for best performance.

After the application is built, it can simply be run via:

```
./target/whatever-1.0.0-SNAPSHOT-runner
```

As an extra thing to try, you can set:

```properties
quarkus.langchain4j.llama3.chat-model.pre-load-in-native=true
```

in `application.properties` which results in the GGUF model being pre-loaded in the native image. This will make inference start immediately when the application is run, but does result in an increase in the binary's size.

**Comment by sebastienblanc** [comment]
@geoand sorry for the newbie question but in dev mode , at runtime I'm getting ` java.lang.NoClassDefFoundError: jdk/incubator/vector/FloatVector` I'm on `24.ea.18-graal` and I start the app ` MAVEN_OPTS='--enable-preview' mvn quarkus:dev` , what am I doing wrong ?

**Comment by geoand** [comment]
For the time being, the dev-mode experience isn't great as you need to set:

 `<jvm.args>--add-modules=jdk.incubator.vector --enable-preview</jvm.args>`

in the `properties` of `pom.xml`.

That's not all however, unless you are using Quarkus from `main` and have also set the `<forceC2>true</forceC2>` property,
your inference will be excrutiatingly slow... 

For those reasons, for the time being I don't recommend using Quarkus Dev with Jlama and Llama3.java

**Comment by sebastienblanc** [comment]
Thx ! Sticking to `java -jar` then for now and thx for `--add-modules=jdk.incubator.vector` I was missing that too.

**Comment by sebastienblanc** [comment]
And also, native will only work on `amd64` right ? Getting a ` Support for the Foreign Function and Memory API is currently available only on the AMD64 architecture.` when trying to compile to native on my `m3`

**Comment by geoand** [comment]
I assume, but that's a question for the GraalVM folks :)

**Pull request merged**